android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -69,16 +69,26 @@ class RNIapModule(
                             promise.safeReject(PromiseUtils.E_NOT_PREPARED, "Unable to auto-initialize connection")
                         }
                     } else {
+                        promise.safeReject(PromiseUtils.E_UNKNOWN, "ensureConnection - incorrect parameter in resolve")
                         Log.i(TAG, "Incorrect parameter in resolve")
                     }
                 },
                 {
-                    if (it.size > 1 && it[0] is String && it[1] is String) {
```

Comment:
When this happens, `it` has only item, so `it.size > 1` is becoming false and it's jumping to else statement. 

Please see the output of `it[0]` below:
![Screen Shot 2023-03-16 at 13 03 08](https://user-images.githubusercontent.com/16003411/225880103-841d9a70-23f1-4fa7-9b59-fd2cd474673f.png)